aqt/archives.py

```diff
@@ -442,11 +442,12 @@ def _get_archives_base(self, name, target_packages):
                 try:
                     extensions_xml_text = self._download_update_xml(extensions_xml_url, True)
                 except ArchiveDownloadError:
-                    # In case _download_update_xml ignores the hash and tries to get the url.
+                    # In case _download_update_xml failed to get the url because of no extension.
```

Comment:
The primary failure when the extension doesn't exist is ChecksumDownloadFailure, which is caught by download_update_xml.  I had to modify the code to simulate getting through that and failing with an ArchiveDownloadError.

Reply:
and I see in #867 the OP had set INSECURE_NOT_FOR_PRODUCTION_ignore_hash: True

So I think the original comment is more accurate.

Reply:
When `_download_update_xml` method failed to download sha256 checksum, it does not throw exception but set `xml_hash = None` and call `get_url`.
The exception is thrown from  `get_url` that try to `Updates.xml` itself.

So when got `ArchiveDownloadError` here, it means that it failed to download `Update.xml` not hash.

Reply:
Assuming Settings.ignore_hash is False download_update_xml catches ChecksumDownloadFailure and RETURNS None.  It doesn't call get_url (silent = True).  This is the normal production behavior.

The OP set ignore_hash to True which behaves as you indicated.